File: pset2/readability/readability.c
// Author: Marko Solajic

#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_words(string text);
int count_letters(string text);
int count_sentences(string text);

int main(void)
{
    // ask for text
    string text = get_string("Text: ");

    // calculate words, letters, sentences
    float words = count_words(text);
    float letters = count_letters(text);
    float sentences = count_sentences(text);

    // the average number of letters (L) and the sentences (S) per 100 words
    float L = (letters / words) * 100.0;
    float S = (sentences / words) * 100.0;

    // use the Coleman-Liau index
    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    // print grade number, and also be careful with 16+ and before grade 1
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

int count_words(string text)
{
    int words = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words + 1;
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: pset6/sentimental-readability/readability.py
# Author: Marko Solajic

from cs50 import get_string


def main():
    text = get_string("Text: ")

    # Counting number of words, letters and sentences
    words, words_list = count_words(text)
    letters = count_letters(text, words_list)
    sentences = count_sentences(text)

    # Average num of letters per 100 words
    L = (letters / words) * 100

    # Average num of sentences per 100 words
    S = (sentences / words) * 100

    # Coleman-Liau index
    index = 0.0588 * L - 0.296 * S - 15.8

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


def count_words(text):
    words_list = text.split()
    words_num = len(words_list)
    return (words_num, words_list)


def count_letters(text, words_list):
    letters_num = 0
    for word in words_list:
        for letter in word:
            if letter.isalpha():
                letters_num += 1
    return letters_num


def count_sentences(text):
    sentences_num = 0
    for char in text:
        if char in ".!?":
            sentences_num += 1
    return sentences_num


if __name__ == "__main__":
    main()
